src/externalized/policy/DefaultGamePolicy.cc

```diff
@@ -1,50 +1,72 @@
-
-	f_draw_item_shadow    = (*json)["draw_item_shadow"].GetBool();
-
-	ms_per_game_cycle     = (*json)["ms_per_game_cycle"].GetInt();
-
-	starting_cash_easy    = (*json)["starting_cash_easy"].GetInt();
-	starting_cash_medium  = (*json)["starting_cash_medium"].GetInt();
-	starting_cash_hard    = (*json)["starting_cash_hard"].GetInt();
-
-	f_drop_everything     = (*json)["drop_everything"].GetBool();
-	f_all_dropped_visible = (*json)["all_drops_visible"].GetBool();
-
-	multiple_interrupts = (*json)["multiple_interrupts"].GetBool();
-
-	gui_extras            = (*json)["gui_extras"].GetBool();
-	extra_attachments = (*json)["extra_attachments"].GetBool();
-
-	enemy_weapon_minimal_status   = (*json)["enemy_weapon_minimal_status"].GetInt();
-
-	ai_better_aiming_choice   = (*json)["ai"]["better_aiming_choice"].GetBool();
-	ai_go_prone_more_often    = (*json)["ai"]["go_prone_more_often"].GetBool();
-	threshold_cth_head        = (*json)["threshold_cth_head"].GetInt();
-	threshold_cth_legs        = (*json)["threshold_cth_legs"].GetInt();
-	enemy_elite_minimum_level = (*json)["enemy_elite_minimum_level"].GetInt();
-	enemy_elite_maximum_level = (*json)["enemy_elite_maximum_level"].GetInt();
-
-	pablo_wont_steal          = (*json)["pablo_wont_steal"].GetBool();
-
-	critical_damage_head_multiplier = (*json)["tactical_head_damage_multiplier"].GetDouble();
-	critical_damage_legs_multiplier = (*json)["tactical_legs_damage_multiplier"].GetDouble();
-	chance_to_hit_maximum     = (*json)["chance_to_hit_maximum"].GetInt();
-	chance_to_hit_minimum     = (*json)["chance_to_hit_minimum"].GetInt();
-
-	imp_attribute_max         = (*json)["imp"]["max_attribute_points"].GetInt();
-	imp_attribute_min         = (*json)["imp"]["min_attribute_points"].GetInt();
-	imp_attribute_bonus       = (*json)["imp"]["bonus_attribute_points"].GetInt();
-	imp_attribute_zero_bonus  = (*json)["imp"]["zero_attribute_points_bonus"].GetInt();
-	imp_pick_skills_directly  = (*json)["imp"]["pick_skills_directly"].GetBool();
+	JsonObjectReader gp = JsonObjectReader(json->GetObject());
+	extra_hotkeys = gp.getOptionalBool("extra_hotkeys", true);
+	can_enter_turnbased = gp.getOptionalBool("can_enter_turnbased");
+	middle_mouse_look = gp.getOptionalBool("middle_mouse_look", true);
+
+	f_draw_item_shadow = gp.getOptionalBool("draw_item_shadow", true);
+	ms_per_game_cycle = gp.getOptionalInt("ms_per_game_cycle", 25);
+
+	starting_cash_easy = gp.getOptionalInt("starting_cash_easy", 45000);
+	starting_cash_medium = gp.getOptionalInt("starting_cash_medium", 35000);
+	starting_cash_hard = gp.getOptionalInt("starting_cash_hard", 30000);
+
+	f_drop_everything = gp.getOptionalBool("drop_everything");
+	f_all_dropped_visible = gp.getOptionalBool("all_drops_visible");
+
+	multiple_interrupts = gp.getOptionalBool("multiple_interrupts");
+
+	enemy_weapon_minimal_status = gp.getOptionalInt("enemy_weapon_minimal_status", 0);
+
+	JsonObjectReader ai = JsonObjectReader(gp.GetValue("ai"));
```

Comment:
This is now the last mandatory field in game.json :)

Reply:
All three groupings/objects are, but since this is the first time things can go wrong, we don't need to care yet. It will become important if we add another grouping.